daemon/src/model/cfd.rs

```diff
@@ -700,14 +707,75 @@ pub enum CfdStateChangeEvent {
     CommitTxSent,
 }
 
+/// Returns the Profit/Loss (P/L) as Bitcoin. Losses are capped by the provided margin
 fn calculate_profit(
-    _intial_price: Usd,
-    _current_price: Usd,
-    _interest_per_day: Decimal,
-    _fee: Usd,
-) -> Result<(Amount, Usd)> {
-    // TODO: profit calculation
-    Ok((Amount::ZERO, Usd::ZERO))
+    initial_price: Usd,
+    current_price: Usd,
+    quantity: Usd,
+    margin: Amount,
+    position: Position,
+) -> Result<(Decimal, Decimal)> {
+    let margin = Decimal::from_f64(margin.as_btc()).context("Expect to be a valid decimal")?;
+
+    let one_through_initial_price = dec!(1)
+        .checked_div(initial_price.0)
+        .context("Division to work")?;
+    let one_through_current_price = dec!(1)
+        .checked_div(current_price.0)
+        .context("Division to work")?;
+
+    // calculate profit/loss (P and L) in BTC
+    let profit_btc = match position {
+        Position::Buy => {
+            // for long: profit_btc = quantity_usd * ((1/initial_price)-(1/current_price))
+            quantity
+                .0
+                .checked_mul(
+                    one_through_initial_price
+                        .checked_sub(one_through_current_price)
+                        .context("Subtraction to work")?,
+                )
+                .context("Multiplication to work")?
+        }
+        Position::Sell => {
+            // for short: profit_btc = quantity_usd * ((1/current_price)-(1/initial_price))
+            quantity
+                .0
+                .checked_mul(
+                    one_through_current_price
+                        .checked_sub(one_through_initial_price)
+                        .context("Subtraction to work")?,
+                )
+                .context("Multiplication to work")?
+        }
+    };
+
+    // loss cannot be more than provided margin
+    let margin_plus_profit_btc = margin
+        .checked_add(profit_btc)
+        .context("Expect addition to work")?;
+
+    let in_percent = if profit_btc.is_zero() {
+        Decimal::ZERO
+    } else {
+        profit_btc.checked_div(margin).context("Division to work")?
+    };
+    if margin_plus_profit_btc < Decimal::ZERO {
+        Ok((
```

Comment:
An early return for either one or the other would be more readable in my opinion.